pkg/wallet/wallet.go

```diff
@@ -63,12 +62,59 @@ func (w *BaseWallet) DecryptMessage(encMessage []byte, toVerKey string) ([]byte,
 
 // PackMessage Pack a message for one or more recipients.
 func (w *BaseWallet) PackMessage(envelope *Envelope) ([]byte, error) {
-	return nil, fmt.Errorf("not implemented")
+	if envelope == nil {
+		return nil, errors.New("envelope argument is nil")
+	}
+	// get keypair from db
+	senderKeyPair, err := w.getKey(envelope.FromVerKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed from getKey: %w", err)
+	}
+
+	var recipients []*[crypto.KeySize]byte
+	for _, verKey := range envelope.ToVerKeys {
+		// decode base58 ver key
+		verKeyBytes := base58.Decode(verKey)
+		// create 32 byte key
+		var recipient [crypto.KeySize]byte
+		copy(recipient[:], verKeyBytes)
+		recipients = append(recipients, &recipient)
+	}
+
+	// encrypt message
+	bytes, err := w.crypter.Encrypt(envelope.Message,
+		crypto.KeyPair{Priv: senderKeyPair.Priv, Pub: senderKeyPair.Pub}, recipients)
+	if err != nil {
+		return nil, fmt.Errorf("failed from encrypt: %w", err)
+	}
+	return bytes, nil
 }
 
 // UnpackMessage Unpack a message.
 func (w *BaseWallet) UnpackMessage(encMessage []byte) (*Envelope, error) {
-	return nil, fmt.Errorf("not implemented")
+	var e authcrypt.Envelope
+	if err := json.Unmarshal(encMessage, &e); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal encMessage: %w", err)
+	}
+	var keysNotFound []string
+	for _, v := range e.Recipients {
+		recipVKeyB58 := v.Header.KID
+		// get keypair from db
+		recipientKeyPair, err := w.getKey(recipVKeyB58)
+		if err != nil {
+			if errors.Is(err, ErrKeyNotFound) {
+				keysNotFound = append(keysNotFound, recipVKeyB58)
+				continue
+			}
+			return nil, fmt.Errorf("failed from getKey: %w", err)
+		}
+		bytes, err := w.crypter.Decrypt(encMessage, crypto.KeyPair{Priv: recipientKeyPair.Priv, Pub: recipientKeyPair.Pub})
```

Comment:
the advantage of using the same struct is you can send it as is here too...

second argument can simply be `recipientKeyPair`, no need to create a new KeyPair instance

Reply:
done